usr/sbin/p11sak/p11sak.c

```diff
@@ -3926,6 +4021,72 @@ static void print_utf8_attr(const char *attr, const CK_ATTRIBUTE *val,
     }
 }
 
+static void print_java_midp_secdom_attr(const char *attr, const CK_ATTRIBUTE *val,
+                                        int indent, bool sensitive)
+{
+    CK_JAVA_MIDP_SECURITY_DOMAIN secdom;
+
+    if (sensitive) {
+        printf("%*s%s: [sensitive]\n", indent, "", attr);
+        return;
+    } else if (val->ulValueLen == 0) {
+        printf("%*s%s: [no value]\n", indent, "", attr);
+        return;
+    }
+
+    secdom = *(CK_JAVA_MIDP_SECURITY_DOMAIN *)(val->pValue);
```

Comment:
You should check if `val->ulValueLen == sizeof(CK_JAVA_MIDP_SECURITY_DOMAIN)` here. Error out if not the desired length.

Reply:
Done.